### PR TITLE
[test] Add OSU Allreduce test

### PR DESCRIPTION
**Pull request by ajocksch**
we might do it more general and summarise alltoall and allreduce to a single check with parameters in osu_tests.py

### CODE REVIEW

File: cscs-checks/microbenchmarks/osu/osu_tests.py [hunk 1]
```diff
@@ -72,6 +72,76 @@ def __init__(self):
...
+
+        self.extra_resources = {
+            'switches': {
+                'num_switches': 1
+            }
+        }
+
+
+@rfm.simple_test
+class FlexAllreduceTest(rfm.RegressionTest):
```
Comment (vkarak) [inline cscs-checks/microbenchmarks/osu/osu_tests.py]: This test is not needed at all. The corresponding Alltoall test was a specific requirement from ops.

File: cscs-checks/microbenchmarks/osu/osu_tests.py [hunk 2]
```diff
@@ -72,6 +72,76 @@ def __init__(self):
...
+            'daint:gpu': {
+                'perf': (20.5, None, 2.0)
+            },
+        }
+        self.num_tasks_per_node = 1
+        self.num_gpus_per_node  = 1
+        if self.current_system.name == 'dom':
+            self.num_tasks = 6
+
+        if self.current_system.name == 'daint':
```
Comment (vkarak) [inline cscs-checks/microbenchmarks/osu/osu_tests.py]:
```suggestion
        elif self.current_system.name == 'daint':
```
And remove the previous blank line.

File: cscs-checks/microbenchmarks/osu/osu_tests.py [hunk 3]
```diff
@@ -72,6 +72,76 @@ def __init__(self):
...
+        self.maintainers = ['RS', 'VK']
+        self.sanity_patterns = sn.assert_found(r'^8', self.stdout)
+        self.perf_patterns = {
+            'perf': sn.extractsingle(r'^8\s+(?P<perf>\S+)',
+                                     self.stdout, 'perf', float)
+        }
+        self.tags = {variant}
+        self.reference = {
+            'dom:gpu': {
+                'perf': (6.0, None, 0.1)
```
Comment (vkarak) [inline cscs-checks/microbenchmarks/osu/osu_tests.py]: Name the performance variable properly, put units and change the required ReFrame version on the top.

File: cscs-checks/microbenchmarks/osu/osu_tests.py [hunk 4]
```diff
@@ -72,6 +72,76 @@ def __init__(self):
         self.tags = {'diagnostic', 'ops'}
 
 
+@rfm.required_version('>=2.14')
+@rfm.parameterized_test(['production'])
```
Comment (vkarak) [inline cscs-checks/microbenchmarks/osu/osu_tests.py]: This test need not be parameterized.

File: cscs-checks/microbenchmarks/osu/osu_tests.py [hunk 5]
```diff
@@ -72,6 +72,76 @@ def __init__(self):
...
+        self.valid_systems = ['daint:gpu', 'dom:gpu']
+        self.descr = 'Allreduce OSU microbenchmark'
+        self.build_system = 'Make'
+        self.build_system.makefile = 'Makefile_allreduce'
+        self.executable = './osu_allreduce'
+        # The -x option controls the number of warm-up iterations
+        # The -i option controls the number of iterations
+        self.executable_opts = ['-m', '8', '-x', '1000', '-i', '20000']
+        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu',
+                                    'PrgEnv-intel']
```
Comment (vkarak) [inline cscs-checks/microbenchmarks/osu/osu_tests.py]: Why we need all these programming environments?

File: cscs-checks/microbenchmarks/osu/osu_tests.py [hunk 6]
```diff
@@ -72,6 +72,76 @@ def __init__(self):
...
+            'perf': sn.extractsingle(r'^8\s+(?P<perf>\S+)',
+                                     self.stdout, 'perf', float)
+        }
+        self.tags = {variant}
+        self.reference = {
+            'dom:gpu': {
+                'perf': (6.0, None, 0.1)
+            },
+            'daint:gpu': {
+                'perf': (20.5, None, 2.0)
```
Comment (vkarak) [inline cscs-checks/microbenchmarks/osu/osu_tests.py]: Is it so large the variation?

Comment (ajocksch) [reply cscs-checks/microbenchmarks/osu/osu_tests.py]: yes, it is a sensitive test and will be slowed down by any other parallel workload on the machine

File: cscs-checks/microbenchmarks/osu/osu_tests.py (hunk 5 above)
Comment (ajocksch) [reply cscs-checks/microbenchmarks/osu/osu_tests.py]: reduced to PrgEnv-gnu only

File: cscs-checks/microbenchmarks/osu/osu_tests.py (hunk 4 above)
Comment (ajocksch) [reply cscs-checks/microbenchmarks/osu/osu_tests.py]: done

File: cscs-checks/microbenchmarks/osu/osu_tests.py (hunk 3 above)
Comment (ajocksch) [reply cscs-checks/microbenchmarks/osu/osu_tests.py]: done

File: cscs-checks/microbenchmarks/osu/osu_tests.py (hunk 2 above)
Comment (ajocksch) [reply cscs-checks/microbenchmarks/osu/osu_tests.py]: done

File: cscs-checks/microbenchmarks/osu/osu_tests.py (hunk 1 above)
Comment (ajocksch) [reply cscs-checks/microbenchmarks/osu/osu_tests.py]: I leave it in for the moment, sometimes the allreduce test is used for reference

Comment (vkarak) [reply cscs-checks/microbenchmarks/osu/osu_tests.py]: For reference of what? This test is just a sanity check and I'm wondering what it provides additionally to the flexible alltoall one.

Comment (vkarak) [reply cscs-checks/microbenchmarks/osu/osu_tests.py]: @ajocksch Apart from this comment, this is ready to be merged. Can you tell me why the flexible version of the test is needed?

**Comment by vkarak** [comment]
@ajocksch The test is failing on both Daint/Dom. Please have a look and run manually.

**Pull request merged**